### ISSUE TITLE
std_detect::detect::features() is not available

### ISSUE THREAD

**Issue by ronnychevalier** [rust-lang/stdarch#1385]
While `std_detect::detect::features()` is declared as `pub`, it is not exposed in `std` and the latest version of `std_detect` available on crates.io (0.1.5, 4 years ago) does not contain this function.
To access this function, one must depend on the git repository.

Would it be possible to release a new version of `std_detect` on crates.io? or expose `std_detect::detect::features()` in `std` like other functions of this crate?

**Comment by thomcc** [comment]
I don't think we want to expose this, and it's not clear that releasing std_detect on crates.io is the right call either (or ever was...)

What's the use case you need it for? Avoiding multiple calls to `std::is_*_feature_detected`?

**Comment by ronnychevalier** [comment]
I have a [cargo subcommand](https://crates.io/crates/cargo-multivers) that builds multiple versions of the same binary, each with a different CPU features set, merged into a single binary. At runtime it runs the optimized version based on the list of CPU features supported by the host.
I rely on `std_detect::detect::features()` to know the CPU features supported at runtime.

**Comment by Amanieu** [comment]
The proper interface to use is the `is_*_features_detected` macros. Anything else is an internal implementation detail and not supported.
